lib/jekyll/readers/post_reader.rb

```diff
@@ -50,7 +50,7 @@ def read_publishable(dir, magic_dir, matcher)
     # Returns klass type of content files
     def read_content(dir, magic_dir, matcher)
       @site.reader.get_entries(dir, magic_dir).map do |entry|
-        next unless entry =~ matcher
+        next unless entry&.match?(matcher)
```

Comment:
Prefer `Regexp#match?` as it is `nil`-compliant. `Regexp#match?(nil)` will not return an error.
```suggestion
        next unless matcher.match?(entry)
```